packages/compiler/server/serverlib.ts

```diff
@@ -412,8 +419,22 @@ export function createServer(host: ServerHost): Server {
     const file = await compilerHost.readFile(getPath(params.textDocument));
     const ast = parse(file, { comments: true });
     const ranges: FoldingRange[] = [];
-    for (let i = 0; i < ast.comments.length; i++) {
-      addRange(ast.comments[i].pos, ast.comments[i].end);
+    let i = 0;
+    while (i < ast.comments.length) {
+      const rangeStart = ast.comments[i].pos;
+      let rangeEnd = ast.comments[i].end;
+      let j = i + 1;
+      while (
```

Comment:
I think this is an unnecessary double loop. I think you can just iterate over each one of the commands using this logic

1. If single line comment, start registering a single line fold(keep track of the start)
   2. Go the next comment, if single line again then do check skipWhiteSpace check with the previous comment and if that match then add to the pending range otherwise add the range to the folding list and start a new pending one
 2. If multi line comment, add folding for multi line and if there is a pending single line comment range from before add that one and reset.
When done add any pending single line comment range